### DISCUSSION TITLE
milvus自增id能被指定值覆盖吗

### DISCUSSION THREAD

**Discussion by buptzouy** [milvus-io/milvus#19475] [Q&A and General discussion]
我尝试指定一个值给id的时候，程序报了如下错误：R{exception=Exception: len(columns) mismatch the len(fields), len(columns): 3, len(fields): 2, status=1, data=null}

**Comment by yhmo** [comment]
不能。
Once a primary key field is defined as **auto_id=True**, you **cannot** specify it when you call insert().
Once a primary key field is defined as **auto_id=False**, you **must** provide it when you call insert().

**Reply by buptzouy** [reply]
> 不能。 Once a primary key field is defined as **auto_id=True**, you **cannot** specify it when you call insert(). Once a primary key field is defined as **auto_id=False**, you **must** provide it when you call insert().

@yhmo 请问，如果primary key的auto_id=False,是否影响查询速度或者有其他影响？

**Reply by yhmo** [reply]
对于int64的pk来说，auto_id=true，里面就会以时间戳为基础生成id并保证唯一性。如果auto_id=false，那么用户输入什么就是什么id，甚至是重复的id也可以。对查询性能应该是没影响的，不过如果用户输入了重复的id，则search()有可能返回相同id的结果，query()则会找到一个个匹配的id就立即返回。
对于varchar的pk来说，auto_id不能为true，否则报错。也就是说，varchar的pk只能由用户来提供，不能自动生成。

**Reply by buptzouy** [reply]
明白了，谢谢

**Reply by Bvbrutal** [reply]
how to make it ture , i mean how to create auto_id

**Reply by yhmo** [reply]
> how to make it ture , i mean how to create auto_id

https://milvus.io/docs/create_collection.md

Just set auto_id=True for the primary key field:
```
    field1 = FieldSchema(name=id_field, dtype=DataType.INT64, is_primary=True, auto_id=True)
    field2 = FieldSchema(name=vector_field, dtype=DataType.FLOAT_VECTOR, dim=dim)
    schema = CollectionSchema(fields=[field1, field2])
```

**Reply by Bvbrutal** [reply]
What is the definition of timestamp, why is it different from my time.time()

**Reply by Bvbrutal** [reply]
Also do you know how to vector search in a range like I only want to search in the vector data I just inserted
